### PR TITLE
[Breaking] Rename cas executable to nativelink

### PR DESCRIPTION
**Pull request by aaronmondal**
This change is [Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/573)

### CODE REVIEW

**Review by aaronmondal — COMMENTED** [review]
+@MarcusSorealheis +@allada 

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/573#-:-NmCltv14RPGwcBgOFeq:b-3thtw4)* status: 0 of 2 LGTMs obtained, and pending CI: Analyze (javascript-typescript), Analyze (python), Bazel Dev / ubuntu-22.04, Cargo Dev / ubuntu-22.04, Local / ubuntu-22.04, Remote / large-ubuntu-22.04, asan / ubuntu-22.04, docker-compose-compiles-nativelink (20.04), docker-compose-compiles-nativelink (22.04), integration-tests (20.04), integration-tests (22.04), pre-commit-checks, publish-image, ubuntu-20.04, ubuntu-20.04 / stable, ubuntu-22.04, ubuntu-22.04 / stable, windows-2022 / stable (waiting on @allada and @MarcusSorealheis)

**Review by adam-singer — APPROVED** [review]
> Reviewed 9 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/573)* status: 0 of 2 LGTMs obtained, and pending CI: Remote / large-ubuntu-22.04 (waiting on @allada and @MarcusSorealheis)

*[a discussion](https://reviewable.io/reviews/TraceMachina/nativelink/573#-NmCuP8n4Z-o-11zK4Jo:-NmCuP8n4Z-o-11zK4Jp:bxqj9lm) (no related file):*
Since this is a breaking change, should we tally an entry into https://github.com/TraceMachina/nativelink/blob/main/CHANGELOG.md now or mark the issue tracker in a way such that next release it is easy to find what breaking changes have been made? Git commit titles work for me, maybe we should codify that (or alternatives) in https://github.com/TraceMachina/nativelink/blob/main/CONTRIBUTING.md

As for `CHANGELOG.md` format, do we need gitsha or just reference ticket id and traverse from there to commit? Personally I do like easy access to gitsha, but also seems like an additional thing to manage in terms of updating a changelog file.

___

**Review by aaronmondal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/573)* status: 0 of 2 LGTMs obtained, and pending CI: Bazel Dev / ubuntu-22.04, Cargo Dev / ubuntu-22.04, Remote / large-ubuntu-22.04, asan / ubuntu-22.04, docker-compose-compiles-nativelink (20.04), docker-compose-compiles-nativelink (22.04), integration-tests (20.04), publish-image, ubuntu-20.04, ubuntu-22.04, ubuntu-22.04 / stable, windows-2022 / stable (waiting on @allada and @MarcusSorealheis)

*[a discussion](https://reviewable.io/reviews/TraceMachina/nativelink/573#-NmCuP8n4Z-o-11zK4Jo:-NmD-JkZEvshXIwpD2s9:bcq3sbj) (no related file):*
<details><summary><i>Previously, adam-singer (Adam Singer) wrote…</i></summary><blockquote>

Since this is a breaking change, should we tally an entry into https://github.com/TraceMachina/nativelink/blob/main/CHANGELOG.md now or mark the issue tracker in a way such that next release it is easy to find what breaking changes have been made? Git commit titles work for me, maybe we should codify that (or alternatives) in https://github.com/TraceMachina/nativelink/blob/main/CONTRIBUTING.md

As for `CHANGELOG.md` format, do we need gitsha or just reference ticket id and traverse from there to commit? Personally I do like easy access to gitsha, but also seems like an additional thing to manage in terms of updating a changelog file.
</blockquote></details>

I think that's a good thing in general but it's not clear to me what a good balance between keeping the changelog updated and small commits is.

Technically we only need to update the changelog on every tag. So we don't *have* to always keep it updated. I think I'm leaning more towards only bumping the changelog on commits that are intended to be tagged commits.

___

**Review by aaronmondal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/573)* status: 0 of 2 LGTMs obtained, and pending CI: Bazel Dev / ubuntu-22.04, Cargo Dev / ubuntu-22.04, Remote / large-ubuntu-22.04, asan / ubuntu-22.04, docker-compose-compiles-nativelink (20.04), docker-compose-compiles-nativelink (22.04), integration-tests (20.04), publish-image, ubuntu-20.04, ubuntu-22.04, ubuntu-22.04 / stable, windows-2022 / stable (waiting on @allada and @MarcusSorealheis)

*[a discussion](https://reviewable.io/reviews/TraceMachina/nativelink/573#-NmCuP8n4Z-o-11zK4Jo:-NmD1dQa5VOnNnuh3S_G:b-c7d2ui) (no related file):*
<details><summary><i>Previously, aaronmondal (Aaron Siddhartha Mondal) wrote…</i></summary><blockquote>

I think that's a good thing in general but it's not clear to me what a good balance between keeping the changelog updated and small commits is.

Technically we only need to update the changelog on every tag. So we don't *have* to always keep it updated. I think I'm leaning more towards only bumping the changelog on commits that are intended to be tagged commits.
</blockquote></details>

Looking into this we'll probably have to make some changes to the existing changelog anyways. What we're currently doing is more or less a copy of the git history. That's explicitly *not* what we're supposed to do for the changelog. Instead, it should be a short, concise, human-readable thing that only contains relevant information on CVEs, migration paths and major changes.

See: https://github.com/coreinfrastructure/best-practices-badge/blob/main/docs/criteria.md#change-control

___

**Review by aaronmondal — COMMENTED** [review]
@allada Is this safe to merge without breaking our existing deployments?

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/573#-:-NmD3jbm1WpOlW-58j_l:b-ehejt1)* status: 0 of 2 LGTMs obtained, and pending CI: docker-compose-compiles-nativelink (20.04), docker-compose-compiles-nativelink (22.04) (waiting on @allada and @MarcusSorealheis)

File: Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ lto = true
 opt-level = 3
 
 [[bin]]
-name = "cas"
+name = "nativelink"
```
Comment (MarcusSorealheis) [inline Cargo.toml:14]: this change got me thinking, does [this comment](https://github.com/TraceMachina/nativelink/pull/573/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542L1-L4) at the top of the file still hold? I feel these may have disappeared.

@allada do you know anything about that?

**Review by MarcusSorealheis — APPROVED** [review]
Otherwise, looks good.

**Review by aaronmondal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/573)* status: 0 of 4 LGTMs obtained, and pending CI: Analyze (javascript-typescript), Analyze (python), Bazel Dev / ubuntu-22.04, Cargo Dev / ubuntu-22.04, Local / ubuntu-22.04, Remote / large-ubuntu-22.04, asan / ubuntu-22.04, docker-compose-compiles-nativelink (20.04), docker-compose-compiles-nativelink (22.04), integration-tests (20.04), integration-tests (22.04), pre-commit-checks, publish-image, ubuntu-20.04, ubuntu-20.04 / stable, ubuntu-22.04, ubuntu-22.04 / stable, windows-2022 / stable (waiting on @adam-singer, @allada, @blakehatch, and @MarcusSorealheis)

___
*[`Cargo.toml` line 19 at r1](https://reviewable.io/reviews/TraceMachina/nativelink/573#gh-1434507145:-NmDALjK0jjso8bZAnul:b-owgmsk) ([raw file](https://github.com/TraceMachina/nativelink/blob/874a815a7a4b9c62962860a2b5f6693233334bf7/Cargo.toml#L19)):*
<details><summary><i>Previously, MarcusSorealheis (Marcus Eagan) wrote…</i></summary><blockquote>

this change got me thinking, does [this comment](https://github.com/TraceMachina/nativelink/pull/573/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542L1-L4) at the top of the file still hold? I feel these may have disappeared.

@allada do you know anything about that?

</blockquote></details>

Done. This was indeed wrong.

**Review by allada — APPROVED** [review]
:lgtm:

> Reviewed 8 of 9 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/573#-:-NmDIsny0ty7rfXp6CbO:bnfp4nl)* status: 1 of 4 LGTMs obtained (waiting on @adam-singer, @blakehatch, and @MarcusSorealheis)

**Review by aaronmondal — COMMENTED** [review]
> Dismissed @MarcusSorealheis from a discussion.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/573)* status: <span>:shipit:</span> complete! 1 of 1 LGTMs obtained

**Review by adam-singer — COMMENTED** [review]
:lgtm:

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/573#-:-NmDOUQW1i2t8wSfu1N1:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 1 LGTMs obtained

*[a discussion](https://reviewable.io/reviews/TraceMachina/nativelink/573#-NmCuP8n4Z-o-11zK4Jo:-NmDOM_98p4aH7bWcCMc:b-ia7vzy) (no related file):*
<details><summary><i>Previously, aaronmondal (Aaron Siddhartha Mondal) wrote…</i></summary><blockquote>

Looking into this we'll probably have to make some changes to the existing changelog anyways. What we're currently doing is more or less a copy of the git history. That's explicitly *not* what we're supposed to do for the changelog. Instead, it should be a short, concise, human-readable thing that only contains relevant information on CVEs, migration paths and major changes.

See: https://github.com/coreinfrastructure/best-practices-badge/blob/main/docs/criteria.md#change-control
</blockquote></details>

TIL on coreinfra github project 

___

**Pull request merged**